builtin/fetch.c

```diff
@@ -958,8 +958,10 @@ static int store_updated_refs(const char *raw_url, const char *remote_name,
 				ref->force = rm->peer_ref->force;
 			}
 
-			if (recurse_submodules != RECURSE_SUBMODULES_OFF)
+			if (recurse_submodules != RECURSE_SUBMODULES_OFF &&
+			    (!rm->peer_ref || !oideq(&ref->old_oid, &ref->new_oid))) {
```

Comment:
@dscho Maybe it should be rm->peer_ref &&?

Reply:
I am actually not familiar with the recurse submodules machinery... Sorry!

Reply:
Maybe @stefanbeller?